templates/Parameters.py.template

```diff
@@ -0,0 +1,137 @@
+# *********************************************************
+#
+# File autogenerated for the ${pkgname} package
+# by the rosparam_handler package.
+# Please do not edit.
+#
+# *********************************************************
+import rospy
+
+param_description = $paramDescription
+
+defaults = {}
+
+for param in param_description:
+    defaults[param['name']] = param['default']
+
+
+class ${ClassName}Parameters(dict):
+    def __init__(self):
+        super(self.__class__, self).__init__(defaults)
+        self.from_param_server()
+
+    __getattr__ = dict.__getitem__
+    __setattr__ = dict.__setitem__
+
+    def from_param_server(self):
+        """
+        Reads and initializes parameters with values from parameter server.
+        Called automatically at initialization.
+        """
+        for k, v in self.iteritems():
+            config = next((item for item in param_description if item["name"] == k), None)
+            if config['constant']:
+                self.test_const_param(k)
+                continue
+
+            self[k] = self.get_param(k, config)
+
+    def from_config(self, config):
+        """
+        Reads parameter from a dynamic_reconfigure config file.
+        Should be called in the callback of dynamic_reconfigure.
+        :param config: config object from dynamic reconfigure
+        """
+        for k, v in config.iteritems():
+            # handle reserved name groups
+            if k == "groups":
+                continue
+            if not k in self:
+                raise TypeError("Element {} of config is not part of parameters.".format(k))
+            self[k] = v
+
+    @staticmethod
+    def test_const_param(param_name):
+        if rospy.has_param("~" + param_name):
+            rospy.logwarn(
+                "Parameter {} was set on the parameter server even though it was defined to be constant.".format(
+                    param_name))
+
+    @staticmethod
+    def get_param(param_name, config):
+        def get_type(type_string):
+            if type_string == 'std::string':
+                return str
+            elif type_string == 'int':
+                return int
+            elif type_string == 'bool':
+                return bool
+            elif type_string == 'float' or type_string == 'double':
+                return float
+            else:
+                raise ValueError()
+
+        full_name = "/" + param_name if config['global_scope'] else "~" + param_name
+        val = rospy.get_param(full_name, config['default'])
```

Comment:
Can you also add the test for an existing param on the server? And set the parameter if it doesn't exist yet?  (PR #22 is merged)

Reply:
Will do